### ISSUE TITLE
Rule 405 (PackageHasRetryRule) false positive on apt update_cache

### ISSUE THREAD

**Issue by robertdebock** [ansible/ansible-lint#405]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version
ansible 2.7.1
  config file = None
  configured module search path = [u'/home/robertdb/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, Oct 15 2018, 15:26:09) [GCC 8.2.1 20180801 (Red Hat 8.2.1-2)]

ansible-lint --version
ansible-lint 3.4.23
```

- ansible installation method: pip
- ansible-lint installation method: pip (Took the rules from this repository: ff0770eedf8cf68f3747e923fc7f78cbc9359535)

# Desired Behaviour
Rule 405 checks of modules to install packages have retries configured.

The apt module can do more than install packages:

```
- name: remove software
  package:
    name: some_package
    state: absent
```

In the above example, no network communication is required, so a retry does not make sense.

# Actual Behaviour (Bug report only)
Linting failes:

```
[405] Remote package tasks should have a retry
tasks/main.yml:3
Task/Handler: remove software
```

**Comment by robertdebock** [comment]
Another false positive situation, when the state is absent.

```
- name: remove software
  package:
    name: some_package
    state: absent
```

**Comment by robertdebock** [comment]
In case of local installations, retries won't help:

```
- name: install software from a local source
  package:
    name: "/tmp/some_package.rpm"
    state: present
```

**Comment by awcrosby** [comment]
The `state: absent` issue, and local installation issue, were addressed in the referenced PRs
